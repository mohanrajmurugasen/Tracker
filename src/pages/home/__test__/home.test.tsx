import { render, fireEvent, screen } from "@testing-library/react";
import Home from "../index";
import renderer from "react-test-renderer";

describe("Counter", () => {
  test("Increment", () => {
    render(<Home />);

    const counter = screen.getByTestId("counter");
    const increment = screen.getByTestId("increment");

    fireEvent.click(increment);

    expect(counter).toHaveTextContent("Counter: 1");
  });

  test("Decrement", () => {
    render(<Home />);

    const counter = screen.getByTestId("counter");
    const decrement = screen.getByTestId("decrement");

    fireEvent.click(decrement);

    expect(counter).toHaveTextContent("Counter: -1");
  });

  test("Snapshot", () => {
    render(<Home />);

    const tree = renderer.create(<Home />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
