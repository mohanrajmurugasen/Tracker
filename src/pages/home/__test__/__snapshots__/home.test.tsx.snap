// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot 1`] = `
<div
  className="counter"
>
  <button
    data-testid="increment"
    onClick={[Function]}
  >
    +
  </button>
  <div
    data-testid="counter"
  >
    Counter: 
    0
  </div>
  <button
    data-testid="decrement"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
